compiler/rustc_codegen_ssa/src/target_features.rs

```diff
@@ -87,12 +87,17 @@ pub(crate) fn from_target_feature_attr(
                     // But ensure the ABI does not forbid enabling this.
                     // Here we do assume that LLVM doesn't add even more implied features
                     // we don't know about, at least no features that would have ABI effects!
-                    if abi_feature_constraints.incompatible.contains(&name.as_str()) {
-                        tcx.dcx().emit_err(errors::ForbiddenTargetFeatureAttr {
-                            span: item.span(),
-                            feature: name.as_str(),
-                            reason: "this feature is incompatible with the target ABI",
-                        });
+                    // We skip this logic in rustdoc, where we want to allow all target features of
+                    // all targets, so we can't check their ABI compatibility and anyway we are not
+                    // generating code so "it's fine".
+                    if !tcx.sess.opts.actually_rustdoc {
+                        if abi_feature_constraints.incompatible.contains(&name.as_str()) {
```

Comment:
isn't this just `&&`?

Reply:
`name` is ` Symbol`. So `&&name` does not work, no.

Reply:
I was commenting on both lines because I meant that this is `&&` the control flow operator:
```rust
if x {
    if y {
    };
};

if x && y {
};
```

Reply:
Oh, that's what you meant.^^

Yeah you are right.

Reply:
Now it's already rolled up so I'll rather not push to this PR.

Reply:
them not being merged turned out oddly fortuitous, as they wound up easier to pick apart while backporting, I suppose.